File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    def dfs(idx, curr):
        if idx >= len(numbers):
            if curr == target:
                return 1
            else:
                return 0
        return dfs(idx + 1, curr + numbers[idx]) +dfs(idx + 1, curr - numbers[idx])
    answer = dfs(0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cs
using System;

public class Solution {
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }
    
    int dfs(int[] numbers, int target, int idx, int sum){
        if(idx == numbers.Length)   return sum == target ? 1 : 0;
        
        return dfs(numbers, target, idx+1, sum + numbers[idx]) +
            dfs(numbers, target, idx+1, sum - numbers[idx]);
        
    }
}